server/server.go

```diff
@@ -1754,7 +1754,7 @@ func (s *Server) StartProfiler() {
 	}
 	hp := net.JoinHostPort(opts.Host, strconv.Itoa(port))
 
-	l, err := net.Listen("tcp", hp)
+	l, err := natsListen("tcp", hp)
 	s.Noticef("profiling port: %d", l.Addr().(*net.TCPAddr).Port)
```

Comment:
This may benefit from it? We do not do PING/PONG over the profile or monitor port.

Reply:
Oops.  This highlights why `natsListen()` is a good name, I will revert this one change (and associated test).